tests/unittests/core/test_apply_func.py

```diff
@@ -359,3 +359,13 @@ class Foo:
     foo = Foo(0)
     result = apply_to_collection(foo, int, lambda x: x + 1, allow_frozen=True)
     assert foo == result
+
+
+def test_apply_to_collection_non_roundtrippable_sequence():
+    class NonRoundtrippableSequence(list):
+        def __init__(self, x: int):
+            super().__init__(range(int(x)))
+
+    val = NonRoundtrippableSequence(3)
+    result = apply_to_collection(val, int, lambda x: x + 1)
+    assert val == result
```

Comment:
Wouldn't you expect `result == [1, 2, 3]`? What's a real example where this should become a no-op?

Reply:
cc: @ap-- ^^

Reply:
Sorry, it seems my reply was in pending state for a month and I had to still click on submit review 🤷